src/anemoi/inference/checkpoint/__init__.py

```diff
@@ -63,3 +67,125 @@ def operational_config(self):
 
         LOG.warning("No operational configuration found. Using default configuration.")
         return {}
+
+    def validate_environment(
+        self,
+        all_packages: bool = False,
+        on_difference: Literal["warn", "error"] = "warn",
+        *,
+        exempt_packages: list[str] | None = None,
+    ) -> bool:
+        """
+        Validate environment of the checkpoint against the current environment.
+
+        Parameters
+        ----------
+        all_packages : bool, optional
+            Check all packages in environment or just `anemoi`'s, by default False
+        on_difference : Literal['warn', 'error'], optional
+            What to do on difference, by default "warn"
+        exempt_packages : list[str], optional
+            List of packages to exempt from the check, by default EXEMPT_PACKAGES
+
+        Returns
+        -------
+        bool
+            True if environment is valid, False otherwise
+
+        Raises
+        ------
+        RuntimeError
+            If found difference and `on_difference` is 'error'
+        ValueError
+            If `on_difference` is not 'warn' or 'error'
+        """
+        train_environment = self.provenance_training
+        inference_environment = gather_provenance_info(full=False)
+
+        # Override module information with more complete inference environment capture
+        import pkg_resources
+
+        module_versions = {
+            pkg.project_name.replace("-", "."): pkg_resources.get_distribution(pkg.project_name).version
+            for pkg in pkg_resources.working_set
+        }
+        inference_environment["module_versions"] = module_versions
+
+        exempt_packages = exempt_packages or []
+        exempt_packages.extend(EXEMPT_PACKAGES)
+
+        invalid_messages = {
+            "python": [],
+            "missing": [],
+            "mismatch": [],
+            "critical mismatch": [],
+            "uncommitted": [],
+        }
+
+        if train_environment["python"] != inference_environment["python"]:
+            invalid_messages["python"].append(
+                f"Python version mismatch: {train_environment['python']} != {inference_environment['python']}"
+            )
+
+        for module in train_environment["module_versions"].keys():
+            if not all_packages and "anemoi" not in module:
+                continue
+            elif module.split(".")[0] in exempt_packages:
```

Comment:
The exemption doesn't work on the anemois, you need another case for namespace packages. 

In `EXEMPT_PACKAGES` you have the anemois with a dash instead of a dot.

Reply:
Addressed in upcoming commit